rpcs3/Emu/RSX/VK/VKHelpers.h

```diff
@@ -1309,16 +1317,36 @@ namespace vk
 			VkSurfaceKHR surface;
 			CHECK_RESULT(vkCreateWin32SurfaceKHR(m_instance, &createInfo, NULL, &surface));
 #elif HAVE_VULKAN
-		
-		vk::swap_chain* createSwapChain(Display *display, Window window, vk::physical_device &dev)
+
+		typedef std::variant<
+			std::pair<Display*, Window>
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+			, std::pair<wl_display*, wl_surface*>
+#endif
+		> VkUnixContext;
```

Comment:
This template can be moved to gsframe header instead as it is not restricted to use with vulkan only and holds general unix window and display properties. Name can be changed to reflect that as well, something like unix_display_context or something along those lines